apps/betterangels-backend/accounts/schema.py

```diff
@@ -73,23 +73,35 @@ def generate_magic_link(self, info: Info, data: MagicLinkInput) -> MagicLinkResp
     @strawberry_django.mutation(extensions=[HasPerm(perms=[ClientProfilePermissions.ADD])])
     def create_client_profile(self, info: Info, data: CreateClientProfileInput) -> ClientProfileType:
         with transaction.atomic():
-            client_profile_data: dict = strawberry.asdict(data)
-            user_data = client_profile_data.pop("user") or {}
-
             user = get_current_user(info)
             permission_group = get_user_permission_group(user)
 
-            client = User.objects.create_client(**user_data)
+            client_profile_data: dict = strawberry.asdict(data)
+            user_data = client_profile_data.pop("user", {})
+            contacts_data = client_profile_data.pop("contacts", [])
+
+            client_user = User.objects.create_client(**user_data)
 
             client_profile = resolvers.create(
                 info,
                 ClientProfile,
                 {
                     **client_profile_data,
-                    "user": client,
+                    "user": client_user,
                 },
             )
 
+            if contacts_data:
```

Comment:
Wouldn't this be simpler as it's own mutation to and and remove contacts?

Reply:
yes. Working on this, I started thinking it may make a lot of sense to have individual _Save_ buttons for each section. Barring that, having the one _Save_ button trigger multiple mutations is also preferable to `update_client_profile` hitting 100+ lines

Reply:
the issue with individual mutations is that creating a client profile would require two round trips. Client contact needs a client profile id, so we'd need to (1) create client profile; (2) use the returned id to create client contact. Maybe that's ok because most operations will be updates, not creates? And I assume most clients would be created quick and dirty, not with the entire profile filled out. What do you think?